### PR TITLE
Set fleet to shards.test

### PR DESCRIPTION
**Pull request by cammellos**
Set fleet to shards.test and remove dependency on status-mobile for fleet definition (there's still some work to be done on syncing to completely remove all the node config/fleet from status-mobile and have everything driven by status-go)

### Behavior

New accounts should have `shards.test` fleet set by default
Migrated accounts should have `shards.test` fleet set by default **if they never set one before**

includes https://github.com/status-im/status-mobile/pull/18007 as related changes

### CODE REVIEW

**Comment by churik** [comment]
@cammellos status.prod fleet is set by default on new account and no shards.test in fleets

IMAGE 2023-12-07 15:16:02

File: src/status_im2/contexts/chat/events.cljs [hunk 1]
```diff
@@ -434,3 +434,11 @@
   {:events [:chat/check-last-chat]}
   [{:keys [db]}]
   {:effects.chat/open-last-chat (get-in db [:profile/profile :key-uid])})
+
+(rf/defn fetch-messages
+  {:events [:chat/fetch-messages]}
+  [_ chat-id]
+  {:json-rpc/call [{:method     "wakuext_fetchMessages"
+                    :params     [{:id chat-id}]
+                    :on-success #()
```
Comment (ilmotta) [inline src/status_im2/contexts/chat/events.cljs]: Some time ago I changed `on-success` to be optional, so you can remove this placeholder function.

File: src/status_im2/subs/profile.cljs [hunk 2]
```diff
@@ -63,6 +63,18 @@
  (fn [{:keys [public-key]}]
    public-key))
 
+(re-frame/reg-sub
+ :profile/webview-debug
+ :<- [:profile/profile]
+ (fn [{:keys [webview-debug]}]
+   webview-debug))
+
+(re-frame/reg-sub
```
Comment (ilmotta) [inline src/status_im2/subs/profile.cljs]: FYI: We have configured zprint to allow the sub ID to be on the same line as `reg-sub`, as well as for `reg-event-fx` and `reg-fx`. However, it's not something we are consistently following in the code yet. Being on the same line is very helpful for conducting project-wide searches, line by line, to, for example, *find a subscription containing word X*. I miss this a few times every week. Granted, we can grep with the surrounding context, but it's one more thing to remember.

File: src/status_im2/contexts/profile/rpc.cljs [hunk 3]
```diff
@@ -1,12 +1,19 @@
 (ns status-im2.contexts.profile.rpc
   (:require
+    clojure.set
     [clojure.string :as string]
     [utils.ens.core :as utils.ens]))
 
+(defn rpc->wakuv2-config
+  [wakuv2-config]
+  (-> wakuv2-config
```
Comment (ilmotta) [inline src/status_im2/contexts/profile/rpc.cljs]: The thread macro is unnecessary with just one function call.

File: src/status_im2/contexts/chat/actions/view.cljs [hunk 4]
```diff
@@ -4,6 +4,19 @@
...
+  [chat-id]
+  (rf/dispatch [:hide-bottom-sheet])
+  (rf/dispatch [:chat/fetch-messages chat-id]))
+
+(defn fetch-messages
+  [chat-id]
+  {:icon                :i/download
+   :right-icon          :i/chevron-right
+   :accessibility-label :chat-fetch-messages
+   :on-press            (partial fetch-messages! chat-id)
```
Comment (ilmotta) [inline src/status_im2/contexts/chat/actions/view.cljs]: Normally in the repo we don't use exclamation mark in functions with side-effects. There's also no guideline for this. In my experience it's a convention that's too hard for a large team to follow and no linter can reliably help in Clojure. There are way too many functions with side-effects in frontend code too, so it can be a chore to stay consistent with the convention.

But here you had to use to avoid the naming conflict right? But inlining on-press is clear and avoid the indirection too:

```clojure
(defn fetch-messages
  [chat-id]
  {:icon                :i/download
   :right-icon          :i/chevron-right
   :accessibility-label :chat-fetch-messages
   :on-press            (fn []
                          (rf/dispatch [:hide-bottom-sheet])
                          (rf/dispatch [:chat/fetch-messages chat-id]))
   :label               (i18n/label :t/fetch-messages)})
```

File: src/status_im/keycard/recovery.cljs [hunk 5]
```diff
@@ -182,26 +181,28 @@
...
-                :keycard-instance-uid instance-uid
-                :key-uid              (address/normalized-hex key-uid)
-                :keycard-pairing      pairing
-                :keycard-paired-on    paired-on
-                :chat-key             whisper-private-key}
-               encryption-public-key
-               {}))))
+              ;; NOTE: Disabling as keycard not functional and this needs to be moved to
+              ;; status-go
+              #_(multiaccounts.create/on-multiaccount-created
```
Comment (ilmotta) [inline src/status_im/keycard/recovery.cljs]: To me it would be best to have an issue, delete all the zombie code and add to the issue's description a link to the revision with the original source (for reference). I think it's convenient to comment code out, but this makes the code confusing and zombie code tends to live for a long time (maybe will not the case here?)

File: src/status_im/fleet/core.cljs [hunk 6]
```diff
@@ -1,17 +1,22 @@
...
   (:require
+    [native-module.core :as native-module]
     [re-frame.core :as re-frame]
     [status-im.multiaccounts.update.core :as multiaccounts.update]
     [status-im.node.core :as node]
-    [status-im2.config :as config]
     [status-im2.constants :as constants]
     [utils.i18n :as i18n]
-    [utils.re-frame :as rf]))
+    [utils.re-frame :as rf]
```
Comment (ilmotta) [inline src/status_im/fleet/core.cljs]: It would be nice to see some of the old code you are touching moving to new namespaces in `status-im2` and following guidelines. Not in this PR of course, but one or two follow-up ones could make a big difference. When we touch old code we have more context to make safer changes, so it's a good opportunity :)

File: src/status_im/fleet/core.cljs [hunk 7]
```diff
@@ -1,17 +1,22 @@
...
+    [utils.transforms :as transforms]))
+
+(def default-fleets (transforms/json->js (native-module/fleets)))
```
Comment (ilmotta) [inline src/status_im/fleet/core.cljs]: It helps during refactoring to know when a var is private or not as we don't need to search it outside the file to be sure it's safer to move it, rename it or refactor it. `default-fleets` and `default-fleet` can be private.

File: src/status_im/fleet/core.cljs [hunk 8]
```diff
@@ -1,17 +1,22 @@
...
+(def default-fleet (.-defaultFleet default-fleets))
```
Comment (ilmotta) [inline src/status_im/fleet/core.cljs]: Apparently, we don't need to write interop code here. Suggestion:

```clojure
(def ^:private default-fleets (transforms/json->clj (native-module/fleets)))
(def ^:private default-fleet (:defaultFleet default-fleets))
(def fleets (->> default-fleets :fleets keys (map keyword)))
```

**Comment by churik** [comment]
@cammellos 
### ISSUE 1: CRs are not received 
Logs for sender: [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13612314/Status-debug-logs.zip)
Logs for receiver: [logs (69).zip](https://github.com/status-im/status-mobile/files/13612304/logs.69.zip)


Also I noticed that display name hasn't been resolved at all

IMAGE 2023-12-08 10:25:58

### ISSUE 2: Error when open advanced > Fleets

IMAGE 2023-12-08 10:26:52
Logs: [logs (70).zip](https://github.com/status-im/status-mobile/files/13612341/logs.70.zip)

File: src/status_im/fleet/core.cljs (hunk 8 above)
Comment (cammellos) [reply src/status_im/fleet/core.cljs]: I was trying to avoid expensive conversions to clojure, since this code is run at startup of the app, and that's where we want to be lean, probably does not matter so much, but it's a place where potentially we could be death by a thousand of paper cuts, but it's probably ok in this case

**Comment by cammellos** [comment]
@ilmotta I think I have addressed the feedback, would you mind having another look please?
thank you!

**Review by ilmotta — APPROVED** [review]
> @ilmotta I think I have addressed the feedback, would you mind having another look please?
thank you!

PR looks good @cammellos, thank you.

The only comment I didn't see addressed so that I can sleep like a baby was https://github.com/status-im/status-mobile/pull/18115/files#r1419797218 😄

File: src/status_im/keycard/recovery.cljs (hunk 5 above)
Comment (cammellos) [reply src/status_im/keycard/recovery.cljs]: oh sorry, missed the comment, I can remove it no problem

File: src/status_im/fleet/core.cljs (hunk 8 above)
Comment (ilmotta) [reply src/status_im/fleet/core.cljs]: @cammellos, a tradeoff you can make here to remove the cost at startup completely is to compute the fleets in a subscription or just make all three vars functions and call `(fleets/fleets)` in `fleet-settings`.

Perhaps this is the best we should do because most users won't navigate or care about the fleet settings, and it's fine there to compute from scratch, the perf hit should be minimal and not impact other parts of the app.

**Comment by churik** [comment]
@cammellos installed builds with https://github.com/status-im/status-mobile/commit/c967d0d4c16a09a2b1439489094b676acbb91aea : 
- issue 2 is gone, shards.test is set by default
- issue 1 is still there; I was not able to receive a contact request at all.

Receiver logs: [logs (71).zip](https://github.com/status-im/status-mobile/files/13614896/logs.71.zip)
Sender logs: [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13614913/Status-debug-logs.zip)

**Comment by churik** [comment]
Message reliability issues are the reasons for failed e2e.
### 1) **Reaction was not received in 2 mins**

Sender: [session step](https://app.eu-central-1.saucelabs.com/tests/9ff83345c19f47a183d7ea16155be13f?auth=fe972ff679108140e2868cd48ac8ced9#109 )(08:47 in tests) 
Geth.log: [test_1_1_chat_message_reaction_geth1.log](https://github.com/status-im/status-mobile/files/13633789/test_1_1_chat_message_reaction_geth1.log)

Receiver: [session step](https://app.eu-central-1.saucelabs.com/tests/70e5caa857e94520babc4a88ed9b0a6a?auth=5e0aa05701d40855b3569f1d6ce107d3#129 )(10:33, still no reaction)
Geth.log: [test_1_1_chat_message_reaction_geth2.log](https://github.com/status-im/status-mobile/files/13633797/test_1_1_chat_message_reaction_geth2.log)

### 2) **All group tests failed due to long accept of CR.**
Within 90 sec after clicking on Accept the user (member_1) still hasn't been added as a contact of admin.

Member_1 (accept the request): [session step](https://app.eu-central-1.saucelabs.com/tests/6313711b61bf4582bbf756a1575a86c1?auth=2470eec76dede13dc0c51f3f4ead444c#66) at 07:59 
Geth.log: [TestGroupChatMultipleDeviceMergedNewUI_geth1 (1).log](https://github.com/status-im/status-mobile/files/13633999/TestGroupChatMultipleDeviceMergedNewUI_geth1.1.log)

Admin (receiver): [session step](https://app.eu-central-1.saucelabs.com/tests/7d4a53ecc7044651953d43c44765bee5?auth=9a6ade972b4f69b47cc45bcd1c2607f3#113) from 8:00 to 09:40 - member_1 is not added to contacts, so there is no message of accepting CR from member_1
Geth.log: [TestGroupChatMultipleDeviceMergedNewUI_geth0.log](https://github.com/status-im/status-mobile/files/13634013/TestGroupChatMultipleDeviceMergedNewUI_geth0.log)


### 3) **Most community tests failed because the joining the open community request was stuck in "Pending" > 60 sec**

Community admin: [session step](https://app.eu-central-1.saucelabs.com/tests/eb946c0ddee94885b04cad0455f97d4a?auth=7dc47e9f37128d6a070eeceabb8592bb#146) (08:46, after this step request should be automatically approved)
Geth.log: [TestCommunityMultipleDeviceMerged_geth0 (3).log](https://github.com/status-im/status-mobile/files/13634153/TestCommunityMultipleDeviceMerged_geth0.3.log)


Member: [session step](https://app.eu-central-1.saucelabs.com/tests/ddaec86baaf547b6b97171723193a3ef?auth=d7ea347d302e8eb5faf90918f9fbf325#102 )(sending Request to join at 09:10, to 10:11 still not joined) 
Geth.log: [TestCommunityMultipleDeviceMerged_geth1 (3).log](https://github.com/status-im/status-mobile/files/13634154/TestCommunityMultipleDeviceMerged_geth1.3.log)

**Comment by churik** [comment]
I couldn't receive any messages in 1-1 chat and in the community that should be fetched from online for > 10 mins
**Prerequisites:**
sender and receiver have 1-1 chat and jined the same community, `shards.test` fleet is set
**Steps:**
- receiver: go to offine
- sender: send 20 messages to 1-1 chat and 50 to community
- receiver: go online after 5 mins


received only part of the message history from community that was marked as "sending" and was resent after receiver got back online;
no messages were received while they were offline (both 1-1 and community)

**UPDATE: eventually messages were fetched > 15 mins and after several attempts of re-login**

Logs attached after messages were fetched.

Sender: [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13637967/Status-debug-logs.zip)
Receiver: [logs (72).zip](https://github.com/status-im/status-mobile/files/13637916/logs.72.zip)

**Comment by J-Son89** [comment]
been testing out this branch and seems okay for me. I can chat 1-1, syncs with desktop etc 👍 In general looks good. Will update if I happen to find any issues

**Review by J-Son89 — APPROVED** [review]

**Comment by cammellos** [comment]
@churik how were the e2e in the last run?

**Comment by cammellos** [comment]
let me update status-go actually

**Comment by churik** [comment]
@cammellos 
https://github.com/status-im/status-mobile/pull/18115#issuecomment-1850086025 were when fleet was set to `status.prod`

Let me re-run tests one more time.
Tell me if I can test this PR again if something has been fixed here

**Comment by churik** [comment]
Most of the tests were failed due to the impossibility of fetching community data while tapping on the community link:
image


Steps: 
- User 1: send a link to the comunity to 1-1 chat (video from [here](https://app.eu-central-1.saucelabs.com/tests/dfbae027586a40e081e5536fc413c810?auth=6b49652d32e1e603af4a7209a4ba39da#128))
- User 2: open link (video from [here](https://app.eu-central-1.saucelabs.com/tests/01f0b883f9a34caea613ea1abfa8b513?auth=8d7817057aa5e04ee40b03bea3467770#86))


Logs (User1):  [TestCommunityMultipleDeviceMerged_geth0 (4).log](https://github.com/status-im/status-mobile/files/13713700/TestCommunityMultipleDeviceMerged_geth0.4.log)

Logs (User2): [TestCommunityMultipleDeviceMerged_geth1 (4).log](https://github.com/status-im/status-mobile/files/13713702/TestCommunityMultipleDeviceMerged_geth1.4.log)

This is reproducible manually too, using + > "Create open community" button) - maybe the method needs to be adjusted on status-go side (not sure)?

**Comment by churik** [comment]
after last changes cannot receive a contact request (only ONCE, could not reproduce manually anymore)

Leaving here comment to preserve the history
Fleet is set to `shards.test`
Sender:
[Status-debug-logs-1.zip](https://github.com/status-im/status-mobile/files/13718738/Status-debug-logs-1.zip)

Receiver: 
[logs (77).zip](https://github.com/status-im/status-mobile/files/13718741/logs.77.zip)

**Comment by churik** [comment]
Found an issue (Android only, tested on Pixel 7A, Android 13) - messages are not resent to the community.
Steps: 
- create a community
- put the device offline 
- send several messages to community
- put the device back online and wait

On IOS messages are resent ~ for 1 min, on Android they are never re-sent.
The issue is not reproducible on current nightly (go v0.171.34, status.prod fleet) and in 1-1 chats.

Logs from Android: 
[Status-debug-logs-2.zip](https://github.com/status-im/status-mobile/files/13719218/Status-debug-logs-2.zip)

Video: 

https://github.com/status-im/status-mobile/assets/4557972/08fb1f4b-d101-4f62-a324-1d0058dec99e

So basically all these messages are stuck in "Sending" state.
